contents/javascript/Javascript-framework-Angular.md

```diff
@@ -0,0 +1,202 @@
+
+1. **Angular supports Single Page Applications.** Single Page Applications are a type of web applications that load a single HTML page, and the page is updated dynamically according to the interaction of the user. SPAs can communicate with the back-end servers without refreshing the full web page, for loading data in the application. They provide a better user experience.
+The fact that Angular supports the development of SPAs, is a good reason to learn the framework.
+ 
+2. **Angular uses TypeScript.** Angular applications are built using TypeScript language, a superscript of JavaScript. It helps catch errors early when writing the code or performing maintenance tasks. This language improves security by providing services like navigation, refactoring, and autocompletion thereby easing the development process for engineers. You can also opt out of its inbuilt features when needed.
+
+TypeScript was created and is maintained by Microsoft, and has been <a href="https://developer.telerik.com/featured/the-rise-of-typescript/">growing in popularity</a> for the last few years; therefore it’s not much of a risk to incorporate TypeScript into your application development. It’s not going anywhere.
+                                                  
+3. **Google Long-Term Support.** Some software engineers consider the fact that Angular is supported by Google an advantage of the technology. While this may sound justified, Google itself is not enough. The good sign though is that Google announced Long-Term Support (LTS) for the technology. Igor Minar and Steven Fuin, the engineers behind Angular, confirmed this commitment in the <a href="https://medium.com/google-developer-experts/my-picks-from-ng-conf-2017-d971842c0d05">ng-conf 2017 Keynote</a>. 
+
+[source](https://hackr.io/blog/why-should-you-learn-angular-in-2018)
+
+### Benefit: Angular provides a declarative user interface.
+
+Angular uses HTML to define the app’s user interface. HTML is a declarative language which is more intuitive and less convoluted than defining the interface procedurally in JavaScript. How does it help? You don’t need to invest your time in program flows and in deciding what loads first. Define what you require and Angular will take care of it. In other words, declarative code enables you to describe what you want done without worrying about how.
+
+Lets see a working example. 
+
+With jQuery you might write the following code to define a textbox that takes numeric values and doesn’t allow alphanumeric input or input that is outside of a defined range.
+
+```
+$("#textbox").numericTextBox({
+    value: 10,
+    min: -10,
+    max: 100,
+    step: 0.75,
+    format: "n",
+    decimals: 3
+});
+```
+
+In the code above, jQuery was instructed to find a DOM element with an id of textbox, then call the extension method numericTextBox and pass in the various parameters that configure the initial value, ranges, etc.
```

Comment:
IMO the explanation is not really needed as the code itself should be self-documenting.